### DISCUSSION TITLE
Select only one row

### DISCUSSION THREAD

**Discussion by reisnobre** [TanStack/table#2151] [General]
Hi.
There's a way to:
1. Select a row clicking on the row instead of an ugly checkbox ?
2. Have only one row selected at the same time ?

**Comment by omridevk** [comment]
1. I solved this by doing this:
by using toggleRowSelected , you get back from useTable()
that way you can add click listener on the row, and update the selected state of the row, see example here:
https://github.com/omridevk/node-cleaner/blob/add_delete_status_for_projects/app/modules/projects/Table.tsx#L102

then usage here:
https://github.com/omridevk/node-cleaner/blob/846f1b996981348eb0fc8a98ec44dfd1a6a7a453/app/modules/projects/Rows.tsx#L66

When clicking the row, I just update the selected state of the row since I now know the row id since I attach the event handler on the row itself, giving you access to the row id.

2. I am not sure what you mean by that? do you want to make sure only one row is selected at a time? 
if so, just on top of my head maybe you can clear all the rowSelectedState when a row is selected, then update it again with the selected row, not sure how performance it will be or how the UX will look. 
I'll try to think of other solution.

**Reply by omridevk** [reply]
for 1.
created this quick sandbox:
https://codesandbox.io/s/wizardly-platform-sthhq?file=/src/App.js

See the changes there. added toggleRowSelected
and called it on the row event listener.

**Reply by omridevk** [reply]
of course I would add css 
```css
cursor: "pointer"
```
on the entire row.

**Comment by dkulakov** [comment]
1. https://codesandbox.io/s/tannerlinsleyreact-table-basic-5wpdk
2. see my case in topic https://github.com/tannerlinsley/react-table/discussions/2126

1, 2 https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/streaming-rows?file=/src/App.js

**Reply by tahv0** [reply]
@dkulakov getting `Object literal may only specify known properties, and 'autoResetSelectedRows' does not exist in type 'TableOptions<object>'`
